### PR TITLE
Setup JS Framework Jest

### PR DESCRIPTION
**Pull request by alexdesi**
### What?
This PR setup Jest, in order to write unit tests for our JS codes.
It adds the first test for the commodities.js (former commodities.js.erb)

### Jira link

HOTT-991

### CODE REVIEW

File: app/webpacker/src/javascripts/commodities.js [hunk 1]
```diff
@@ -2,13 +2,16 @@
 /*jslint
  white: true */
 
-import IMask from 'imask';
-import debounce from "./debounce";
+// import IMask from 'imask';
```
Comment (willfish) [inline app/webpacker/src/javascripts/commodities.js:5]: Do we not use this import?

File: babel.config.js [hunk 2]
```diff
@@ -20,9 +20,14 @@ module.exports = function(api) {
       isTestEnv && [
         require('@babel/preset-env').default,
         {
-          targets: {
-            node: 'current'
-          }
+          // targets: {
```
Comment (willfish) [inline babel.config.js]: Probably we should remove rather than comment this?

File: spec/javascript/setup.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+var jsdom = require('jsdom');
+global.$ = require('jquery')(new jsdom.JSDOM().window);
+
```
Comment (willfish) [inline spec/javascript/setup.js]: https://forum.sublimetext.com/t/remove-any-extra-blank-lines-at-end-of-file-on-save/6175/3

File: app/webpacker/src/javascripts/commodities.js (hunk 1 above)
Comment (alexdesi) [reply app/webpacker/src/javascripts/commodities.js:5]: that's a tricky topic,
the only way we were able to make Jest working was by using the commonjs module.
Also, the Jest docs states: 

> Jest ships with experimental support for ECMAScript Modules (ESM).

Comment (alexdesi) [reply app/webpacker/src/javascripts/commodities.js:5]: https://jestjs.io/docs/ecmascript-modules

**Review by willfish — APPROVED** [review]

**Pull request merged**